### DISCUSSION TITLE
Difference between browser.waitUntil and $(selector).waitUntil()

### DISCUSSION THREAD

**Discussion by prakashpqa** [webdriverio/webdriverio#9566] [Q&A]
I am not able to understand the difference between **browser.waitUntil** and **$(selector).waitUntil()**

Especially this statement is confusing me.

`As opposed to other element commands WebdriverIO will not wait for the element to exist to execute this command.`

Ref: https://webdriver.io/docs/api/element/waitForExist

Please help.

**Comment by christian-bromann** [comment]
> I am not able to understand the difference between **browser.waitUntil** and **$(selector).waitUntil()**

Technically there are none.

> Especially this statement is confusing me.

Commands like `click` require an element to be existing. You can e.g. do this `await $("foobar").click()` and the click is only executed once `foobar` was found. This is part of WebdriverIOs automatic retry strategy. An exception for this behavior is `waitForExist` for obvious reasons.

Does it make more sense now?
